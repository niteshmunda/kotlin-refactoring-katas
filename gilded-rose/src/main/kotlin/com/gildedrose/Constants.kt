package com.gildedrose

object Constants {
  const val AGED_BRIE = "Aged Brie"
  const val BACKSTAGE_PASSES = "Backstage passes to a TAFKAL80ETC concert"
  const val SULFURAS_HAND_OF_RAGNAROS = "Sulfuras, Hand of Ragnaros"
  const val QUALITY_LOWER_BOUND = 0
  const val QUALITY_UPPER_BOUND = 50
  const val QUALITY_CONSTANT_ONE = 1
  const val QUALITY_CONSTANT_ELEVEN = 11
  const val QUALITY_CONSTANT_SIX = 6
}
