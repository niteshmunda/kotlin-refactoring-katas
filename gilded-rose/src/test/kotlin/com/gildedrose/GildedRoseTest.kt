package com.gildedrose

import com.gildedrose.Combinations.dotProduct
import com.gildedrose.Constants.AGED_BRIE
import com.gildedrose.Constants.BACKSTAGE_PASSES
import com.gildedrose.Constants.SULFURAS_HAND_OF_RAGNAROS
import org.approvaltests.Approvals
import org.approvaltests.namer.NamerFactory
import org.approvaltests.reporters.QuietReporter
import org.approvaltests.reporters.UseReporter
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.MethodSource
import java.io.ByteArrayOutputStream
import java.io.PrintStream

class GildedRoseTest {

  private val interactions = Interactions()
  companion object {
    @JvmStatic
    fun combinationApprovalParameters(): List<Triple<String, Int, Int>> {
      val allNames = allNames()
      val allSellInRange = allSellInRange()
      val allQualityRange = allQualityRange()

      val allNamesDotAllSellInRange = dotProduct(
        items1 = allNames.toTypedArray(),
        items2 = allSellInRange.toTypedArray(),
        combine = { name, sellIn -> name to sellIn }
      )

      return dotProduct(
        items1 = allNamesDotAllSellInRange.toTypedArray(),
        items2 = allQualityRange.toTypedArray(),
        combine = { (name, sellIn), quality -> Triple(name, sellIn, quality) }
      ).distinct()
    }

    private fun allQualityRange(): List<Int> {
      return listOf(-1, 0, 2, 6, 10, 11, 49, 50, 65)
    }

    private fun allSellInRange(): List<Int> {
      return listOf(0, 12, 50)
    }

    private fun allNames(): List<String> {
      return listOf(AGED_BRIE, SULFURAS_HAND_OF_RAGNAROS, BACKSTAGE_PASSES, "foo")
    }
  }

  @Test
  fun foo() {
    val items = arrayOf(Item("foo", 0, 0))
    val app = GildedRose(items)
    app.updateQuality(items)
    assertEquals("foo", app.items[0].name)
  }

  @ParameterizedTest
  @MethodSource(value = ["combinationApprovalParameters"])
  @UseReporter(QuietReporter::class)
  internal fun `combination approval test for regression protection`(
    input: Triple<String, Int, Int>
  ) {
    // given
    val (name, sellin, quality) = input
    val testName = fileName(input)

    // when
    val app = GildedRose(arrayOf(Item("foo", 0, 0)))// temporary for just object creation
    val output = app.fwdUpdateInformation(Item(name,  sellin, quality))

    interactions.append("result(out)", output.toString())
    // then

//    CombinationApprovals.verifyAllCombinations(
//      Function1 {
//        app.fwdUpdateInformation(Item(name,  sellin, quality))
//      }
//    )

    NamerFactory.withParameters(testName).use {
      Approvals.verify(textWriter(interactions.toString()), NamerWithoutBuildInfo())
    }
  }

  @Test
  fun `testing using main block for 2 days`() {

    // given
    val days = "2"
    val outputStream = ByteArrayOutputStream()

    // when
    System.setOut(PrintStream(outputStream))
    main(arrayOf(days))

    // then
    Approvals.verify(outputStream.toString())
  }

  @Test
  fun `testing using main block for 50 days`() {

    // given
    val days = "50"
    val outputStream = ByteArrayOutputStream()

    // when
    System.setOut(PrintStream(outputStream))
    main(arrayOf(days))

    // then
    Approvals.verify(outputStream.toString())
  }

  private fun fileName(input: Triple<String, Int, Int>): String {
    val (name, sellin, quality) = input
    return "-$name-$sellin-${quality}"
  }
}
