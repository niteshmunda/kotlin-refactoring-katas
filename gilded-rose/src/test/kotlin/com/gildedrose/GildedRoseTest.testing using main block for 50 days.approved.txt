OMGHAI!
-------- day 0 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 20
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 7
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 6

-------- day 1 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 19
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 6
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 5

-------- day 2 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 18
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 5
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 4

-------- day 3 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 17
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 4
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 3

-------- day 4 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 16
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 3
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 2

-------- day 5 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 15
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 2
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 1

-------- day 6 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 14
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 1
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 7 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 13
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 8 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 12
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 9 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 11
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 10 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 10
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 11 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 9
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 12 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 8
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 13 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 7
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 14 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 6
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 15 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 5
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 16 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 4
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 17 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 3
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 18 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 2
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 19 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 1
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 20 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 21 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 22 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 23 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 24 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 25 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 26 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 27 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 28 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 29 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 30 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 31 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 32 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 33 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 34 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 35 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 36 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 37 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 38 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 39 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 40 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 41 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 42 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 43 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 44 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 45 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 46 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 47 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 48 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 49 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

-------- day 50 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 0
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 0
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 0

